compiler/desugared/ast.ml

```diff
@@ -40,7 +40,7 @@ module LabelSet : Set.S with type elt = LabelName.t = Set.Make (LabelName)
 module ScopeDef = struct
   type t =
     | Var of ScopeVar.t * StateName.t option
-    | SubScopeVar of SubScopeName.t * ScopeVar.t
+    | SubScopeVar of SubScopeName.t * ScopeVar.t * Pos.t
```

Comment:
To what does the `Pos.t` points to ? And why only for `SubScopeVar` and not for `Var` ?

Reply:
`ScopeVar.t` / `StateName.t` already contain a position; `SubScopeName.t` as well. But in this case, the position of the dot `subscope.var` can be different from both the above.